### PR TITLE
Spess station massacre (Dismemberment)

### PR DESCRIPTION
**Pull request by RemieRichards**
- Adds Dismemberment. (Port of my NT dismemberment + Core code rewrite, yes I'm insane to rewrite this every single time)
- Humans are now made of overlays (They maintain an icon_state at round start due to getFlatIcon calls)
- There is now a limb_icon_cache so that humans don't need to always redraw all their limb overlays
- Humans no longer have a base_icon_state var
- A bunch of human icon indices are now in the DEFINES (as with @Razharas mutation PR, we're going to conflict but it's necessary)

Remaining issues/TODO:
- [ ] Hook into Razharas' mutations rewrite
- [x] Newer species icons weirdness
- [x] Admin heal proc does not regenerate limbs yet
- [x] Move SHARP flags to appropriate weapons
- [x] Gun dismemberment (Physical bullets only)
- [x] Explosion dismemberment
- [ ] General polishing
- [x] Add the NTstation deusexmachina admin verb which allows you to change limb status (Organic, Robotic)
- [x] <B>Stretch Goal</B>: Greyscale humans, this improves the code quite a bit, but I'm reliant on WJ figuring out some colouring and sprite bases.
- [ ] <B>Stretch Goal</B>: Make drop_limb() not racist (Difficult, but I'll try)
- [ ] <B>Stretch Goal</B>: Changeling limb cool stuff

TESTING:
Spawn a weapon with the SHARP flag and high force (There's a special spawnable katana) and lop off some limbs!

Will remove 'DNM' tag once It's ready.

### CODE REVIEW

**Comment by RemieRichards** [comment]
Just a heads up to all, the OP has changed, it now states that this PR is licensed under GPL not our new AGPL license. this re-licensing was due to a request to use this, or parts of this, PR.

the people who need to understand this do.

**Comment by N3X15** [comment]
Actually, we were just asking if you'd license it to /vg/ under GPL, since we have a different license and needed to get your express permission to do so before yoinking it.  You don't have to change the license for TG, since your dealings with us are entirely seperate. (And TG requires AGPL, so don't change your PR for them, it was fine as-is.)

**Comment by RemieRichards** [comment]
@N3X15 I was speaking with D3athrow since I didn't know how to proceed, we both thought just a message on the PR would do, how will you guys obtain my code, just nab from the branch? (since it's all working, all conflicts here are with tg)

**Comment by N3X15** [comment]
It's 3AM and I'm talking like a retard, so, for clarification, you now have two license agreements:
 - TG, AGPL (You agreed to that by making the PR)
 - /vg/, GPL (since we asked you and you were okay with that)

We can just nab the code from here, no need to run around like a headless chicken.  Of course, I wouldn't need to do any *work* if you made a PR for us, too...

**Comment by RemieRichards** [comment]
@N3X15 I'll keep this one AGPL, you guys can take it when you need it, and if I have time today I'll try to stick a PR up.

**Comment by N3X15** [comment]
Great.  I've been up to my eyeballs in cmake scripts all day and I need some sleep, so I'll muck around with it tomorrow.

SORRY FOR SHITTING UP THE PR, FOLKS

**Comment by paprka** [comment]
N3X15 stealing code /vg/ conspiracy confirmed

**Comment by paprka** [comment]
When

**Comment by Steelpoint** [comment]
How goes this PR?

**Comment by Jalleo** [comment]
Remie is currently helping the remake quite a bit at the moment so this is on Hiatus for a short duration at least

**Comment by GunHog** [comment]
RR COME BACK!!

**Comment by RemieRichards** [comment]
come back? where do you think I am...

**Comment by GunHog** [comment]
Come back to this PR! Finish it already! It is literally from *last year*!

**Comment by Tenebrosity** [comment]
>When

**Comment by Fox-McCloud** [comment]
Personally, I'd raise the maximum health of arms from 75 to 120-150 if you're going to do this.

15% chance to dismember after a mere 3 hits with a cleaver (or 24%/30% chance after 2 hits with the fireaxe/esword) is pretty ridiculous, in my opinion.

**Comment by Dorsisdwarf** [comment]
Eswords should send limbs everywhere when you go on a murderspree if you ask me, especially dual-eswords.

Lightsabers be dangerous man

**Comment by Tenebrosity** [comment]
Yeah, but from a balance perspective that's even more ggnore.

**Comment by Dorsisdwarf** [comment]
Yeah but that makes it even funnier when the clown slips the dual-esworder and cuts just one arm off and runs away with the arm, so the dual-sworder cannot into dual swords.

**Comment by WJohn** [comment]
One day.

**Comment by GunHog** [comment]
RR, come on. Finish this. Stop making us wait!

**Comment by RemieRichards** [comment]
@GunHog Do you realise how demoralising it is to create something 3 times, each time going to significant effort to ensure the next iteration is far superior to the last, to yet AGAIN be blocked/delayed by something else? (in this case communication between this system and another)

**Comment by GunHog** [comment]
O-oh, okay, RR. I was going to make a joke about giving you more time, but I cannot, knowing that feeling with Botcall. I am sorry.

**Comment by RemieRichards** [comment]
@GunHog don't worry about it buddy, If you didn't badger me I probably WOULD have forgotten about and given up on it by now ;)

**Comment by Iamgoofball** [comment]
don't believe the lies

**Comment by Jalleo** [comment]
@RemieRichards i think people would be fine to have a you are conflicting with rr this frozen thing

**Comment by dannno** [comment]
lmao does anyone ever understand what jalleo is saying

**Comment by Jalleo** [comment]
What I was meaning when you next try this people would probably be willing to have a mini freeze to prevent any conflicts that could occur during your development of dismemberment to get it merged.

**Comment by Lumipharon** [comment]
Please finish this so I can make my secret maint lair and fill it with dozens of legless victims.

**Comment by Dorsisdwarf** [comment]
While trawling the issue reports for bad reports to flag up, I found a statement where you said it was coming out "wednesday to friday".

Last november.

:(

**Comment by RemieRichards** [comment]
@Dorsisdwarf it's not a well hidden secret that this build of dismemberment was 100% complete.
Yes, that's right, it WAS complete, it conflicts with Razharas's datum mutation system, and when I tried to update to get the two to work nicely I couldn't and got demoralised, haven't touched it since.

Dismemberment will remain on my todo list until it is a feature of tgstation13 or the feature is no longer wanted.

**Comment by Miauw62** [comment]
it will always be wanted.

**Comment by Jalleo** [comment]
I want to have bags full of arms

**Comment by Dorsisdwarf** [comment]
Do you know which bits it actually conflicts with? 
Working with raz to help you glue the bits together might help.

**Comment by RemieRichards** [comment]
@Dorsisdwarf it's his mutation icon drawing code.
I just need to set some time a side to mash his with mine.

When's Easter? it's soon right? I have a break then so MAYBE I'll eat some eggs and rewrite some shit and get this done (although saying that I'll probably end up rewriting the whole thing again because I'm insane)
(Probably remove gun dismemberment too since by the looks of things not many people agree with it and I'm not tied to the idea.)

**Comment by RemieRichards** [comment]
Also I swear to god if you guys over-hype this I'll kill you all and take your limbs as trophies.

**Comment by MMMiracles** [comment]
You can only take our limbs if dismemberment gets finished though.

**Comment by GunHog** [comment]
> Also I swear to god if you guys over-hype this I'll kill you all and take your limbs as trophies.

You have to get it merged, first! Honk honk!

**Comment by Jalleo** [comment]
Insert grumpy runtime meme here to do with medbay and body parts.

**Comment by AlexCrimson** [comment]
Easter Dismemberment HYPE?!

**Comment by Iamgoofball** [comment]
goddamn it remie

**Comment by RemieRichards** [comment]
My easter holiday has started, so yeah, I'll get my 2nd Ninja PR out of the way and then look at dismemberment again,

**Comment by tedward1337** [comment]
Happening?????

**Comment by CorruptComputer** [comment]
When?

**Comment by RemieRichards** [comment]
I SAID EASTER

**Comment by GunHog** [comment]
DO NOT HYPE US, RR. DO NOT DO IT!

**Comment by RemieRichards** [comment]
NO I TOLD YOU GUYS NOT TO HYPE THIS, IT'LL BE YOUR FAULT

**Comment by Jalleo** [comment]
I aint saying nothin I dont want a runaway train those wreck timetables.

**Comment by tedward1337** [comment]
EASTER WAS YESTERDAY
COME ON RR

**Comment by AlexCrimson** [comment]
Wow. Cant believe you hyped us up like this then just let us down. So cruel.

**Comment by dannno** [comment]
100th comment

**Comment by RemieRichards** [comment]
I'm sorry you guys cant read.
I said I'd work on it over easter.

Easter for me is my entire holiday off, I have a week still.
But maybe I shouldn't even bother if you're just going to act entitled.

**Comment by Dorsisdwarf** [comment]
plz no don't be scared by the shitposters

**Comment by AlexCrimson** [comment]
Well i was just joking, but whatever.

**Comment by Razharas** [comment]
Orthodox(the only true one and god-approved) easter has come

**Comment by Dorsisdwarf** [comment]
*fearful glance*

**Comment by Cheridan** [comment]
:^)

**Comment by Jalleo** [comment]
Just leave it be. It will be done when it will be done. Dont add pressure where none is needed.

**Comment by Dorsisdwarf** [comment]
Why is it a PR if it isn't dooooone

**Comment by demonfiren** [comment]
Oh, man, I dismember this.

**Comment by RocKeD** [comment]
"Stretch Goal: Make drop_limb() not racist (Difficult, but I'll try)"
I laughed pretty hard.

**Comment by RemieRichards** [comment]
What can I say, I have a good sense of humour.

**Comment by GunHog** [comment]
It will help the hype last the year it is going to take to finish this :P

**Comment by optimumtact** [comment]
The number of people who commented on this PR who are no longer around is high

**Comment by freerealestate** [comment]
plz

**Comment by GunHog** [comment]
You have yet to realize that this project is dead, and that may never be finished!

**Comment by RemieRichards** [comment]
If it's dead I should probably close this.

**Comment by palpatine213** [comment]
Remie should probably have her feature privileges revoked until she finishes this.

**Comment by duncathan** [comment]
That's a really stupid suggestion

**Comment by palpatine213** [comment]
It was a joke

**Comment by RocKeD** [comment]
What's this thing called a joke?

**Comment by Razharas** [comment]
By the power of being an ultimate dick i close this PR
RR can open it any moment they decide to return to it

**Comment by freerealestate** [comment]
ded

**Comment by Jalleo** [comment]
Pfft everyone is buthurt seriously a good feature needs time to mature to be a lovely one. Just like a good drink of whatever you like (of which i would probably not like i dont drink)

**Comment by Lumipharon** [comment]
I remember some talk about greyscales or something, but honest question, why was dismemberment done somewhat easily on NT, but it's run into such issues here?

**Comment by RemieRichards** [comment]
@Lumipharon Razharas went ahead with his mutation overhaul (The one that broke genetics for weeks) and decided to handle all the overlays and sprites in a way 100000% incompatible with the dismemberment I'd written at the time (Which handled 90% of human overlays and sprites, making his a snowflake change) I tried to merge the two once, I have never tried since as it immediately demoralised me, that and everyone's constant nagging (Read: Dorsisdwarf)

tl;dr Raz pulled the rug out from under me+dismemberment

**Comment by Lumipharon** [comment]
Well tht sucks. Dismemberment was great - does the mutaion thing even functionally do anything different?

**Comment by optimumtact** [comment]
it changes how sprites for characters are built

**Comment by CorruptComputer** [comment]
If this ever gets merged I'll write a real GUI for MapMerge, @Aranclanos was a witness to this statement.

**Comment by MrStonedOne** [comment]
F

**Comment by GunHog** [comment]
Welp, we are never getting THAT either, then.

**Comment by optimumtact** [comment]
F

**Comment by Shadowlight213** [comment]
F

**Comment by Jalleo** [comment]
Just stop nagging someone else is trying to do it now.
What do you want a chocolate coin?

**Comment by freerealestate** [comment]
Well, I wouldn't mind one...

**Comment by optimumtact** [comment]
I'll have one too thanks

**Comment by GunHog** [comment]
I would like one as well, please!

**Comment by Jalleo** [comment]
Okay here's the mold
Image of Coin Mold

**Comment by tedward1337** [comment]
F

**Comment by Crystalwarrior** [comment]
any chance you could tell me where the grayscale icons for humans are @RemieRichards ?

**Comment by Crystalwarrior** [comment]
edit: nvm it seems like they're not in the PR but comments instead, huh.

**Comment by PKPenguin321** [comment]
F

**Comment by Shadowlight213** [comment]
M

**Comment by TechnoAlchemisto** [comment]
cactus

**Comment by PKPenguin321** [comment]
Cactus?

**Comment by TechnoAlchemisto** [comment]
yeah, they're prickly but some taste good

**Comment by PKPenguin321** [comment]
oh

**Comment by Shadowlight213** [comment]
Neat. What's the best way to prepare cactus?

**Comment by TechnoAlchemisto** [comment]
theyre pretty good raw, but i've actually had them grilled before, you can also use them as a slightly sweet accentuation on some beef dishes.
